lib/samson/hooks.rb

```diff
@@ -63,6 +110,32 @@ def render_views(name, view, *args)
         nil
       end
 
+      def plugin_setup
+        Samson::Hooks.plugins.
+          each(&:require).
+          each(&:precompile_assets).
+          each(&:add_lib_path).
+          each(&:add_decorators)
+      end
+
+      def render_javascripts(view)
+        Samson::Hooks.plugins.map(&:javascripts).flatten.each do |script|
+          view.instance_exec do
+            concat view.javascript_include_tag(view.asset_path(script))
```

Comment:
the more plugins we get the more sandboxing will be nice ... but not sure how much work that will be, also means we lose a lot of flexability